app/renderer/renderer.py

```diff
@@ -13,6 +13,31 @@ def __init__(self, schema, response_store, validation_store, navigator, metadata
         self._navigator = navigator
         self._metadata = metadata
 
+        # Create an object from a dict
+        class ContextEntry:
+            def __init__(self, entries):
+                self.__dict__ = entries
```

Comment:
You have exactly the same class called ObjectFromDict in test_plumber.py. Maybe refactor both into the app.utilities package.